File: library/src/test/java/me/proxer/library/api/ConferenceAdapterTest.java
package me.proxer.library.api;

import me.proxer.library.api.ConferenceAdapter.IntermediateConference;
import me.proxer.library.entitiy.messenger.Conference;
import org.junit.Before;
import org.junit.Test;

import java.util.Date;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Ruben Gees
 */
public class ConferenceAdapterTest {

    private ConferenceAdapter adapter;

    @Before
    public void setUp() {
        adapter = new ConferenceAdapter();
    }

    @Test
    public void testFromJson() {
        assertThat(adapter.fromJson(constructIntermediateConference(true)))
                .isEqualTo(constructConference(true));
    }

    @Test
    public void testFromJsonWithNoImage() {
        assertThat(adapter.fromJson(constructIntermediateConference(false)))
                .isEqualTo(constructConference(false));
    }

    private IntermediateConference constructIntermediateConference(final boolean withImage) {
        IntermediateConference result = new IntermediateConference();

        result.id = "123";
        result.topic = "something";
        result.customTopic = "somethingElse";
        result.participantAmount = 13;
        result.group = true;
        result.read = false;
        result.date = new Date(123456L);
        result.unreadMessageAmount = 2;
        result.lastReadMessageId = "321";
        result.image = withImage ? "avatar:http://example.com/image.png" : "";

        return result;
    }

    private Conference constructConference(final boolean withImage) {
        return new Conference("123", "something", "somethingElse", 13,
                withImage ? "http://example.com/image.png" : "", withImage ? "avatar" : "", true, false,
                new Date(123456L), 2, "321");
    }
}

File: library/src/test/kotlin/me/proxer/library/internal/adapter/ConferenceAdapterTest.kt
package me.proxer.library.internal.adapter

import me.proxer.library.entity.messenger.Conference
import me.proxer.library.internal.adapter.ConferenceAdapter.IntermediateConference
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.Test
import java.util.Date

/**
 * @author Ruben Gees
 */
class ConferenceAdapterTest {

    private val intermediateConferenceWithImage = IntermediateConference(
        id = "123", topic = "something", customTopic = "somethingElse", participantAmount = 13, isGroup = true,
        isRead = false, date = Date(123456L), unreadMessageAmount = 2, lastReadMessageId = "321",
        rawImage = "avatar:http://example.com/image.png"
    )

    private val conferenceWithImage = Conference(
        id = "123", topic = "something", customTopic = "somethingElse", participantAmount = 13,
        image = "http://example.com/image.png", imageType = "avatar", isGroup = true, isRead = false,
        date = Date(123456L), unreadMessageAmount = 2, lastReadMessageId = "321"
    )

    private val intermediateConferenceWithoutImage = intermediateConferenceWithImage.copy(rawImage = "")
    private val conferenceWithoutImage = conferenceWithImage.copy(image = "", imageType = "")

    private val adapter = ConferenceAdapter()

    @Test
    fun testFromJson() {
        adapter.fromJson(intermediateConferenceWithImage) shouldBeEqualTo conferenceWithImage
    }

    @Test
    fun testFromJsonWithoutImage() {
        adapter.fromJson(intermediateConferenceWithoutImage) shouldBeEqualTo conferenceWithoutImage
    }
}
